app/models/authentication.rb

```diff
@@ -158,20 +158,14 @@ def set_tenant_from_group
     self.tenant_id = miq_group.tenant_id if miq_group
   end
 
-  def set_credentials_changed_on
-    return unless @auth_changed
```

Comment:
What if we change this to `return unless password_changed? || auth_key_changed? || username_changed? || ...`?  I'm not a big fan of implementing this in the STI subclasses for Authentications.

Reply:
I was just going with what we've got, but maybe that's the way to go.

Reply:
But, actually, it would still always return true with the `before_save` hook there. We still need to change it to `before_update`.

Reply:
@bdunne I ended up adopting your suggestion. It actually made life easier, as it avoided unexpected results from subclasses.